src/Agent.Worker/Build/TeeTFCommandManager.cs

```diff
@@ -78,12 +78,12 @@ public string ResolvePath(string workspace, string serverPath)
             return localPath;
         }
 
-        public async Task ShelveAsync(string directory, string shelveset, string commentFile)
+        public async Task ShelveAsync(string shelveset, string directory, string commentFile)
         {
-            ArgUtil.NotNullOrEmpty(directory, nameof(directory));
             ArgUtil.NotNullOrEmpty(shelveset, nameof(shelveset));
+            ArgUtil.NotNullOrEmpty(directory, nameof(directory));
             ArgUtil.NotNullOrEmpty(commentFile, nameof(commentFile));
-            await RunPorcelainCommandAsync(FormatFlags.OmitCollectionUrl, "shelve", "-replace", "-recursive", $"-comment:@{commentFile}", shelveset, directory);
+            await RunPorcelainCommandAsync(FormatFlags.OmitCollectionUrl, "shelve", "-saved", "-replace", "-recursive", $"-comment:@{commentFile}", shelveset, directory);
```

Comment:
I was wrong! Adding this "-saved" arg fixes work item association for gated. Alex helped me figure it out.